bravado/mapping/param.py

```diff
@@ -128,6 +132,73 @@ def marshal_param(param, value, request):
             format(location))
 
 
+def unmarshal_param(param, request):
+    """Unmarshal the given parameter from the passed in request like object.
+
+    :type param: :class:`bravado.mapping.param.Param`
+    :type request: :class:`bravado.mapping.request.RequestLike`
+    """
+    param_spec = get_param_type_spec(param)
+    location = param.location
+    cast_param = partial(cast_request_param, param_spec['type'], param.name)
+
+    if location == 'path':
+        raw_value = cast_param(request.path.get(param.name, None))
+    elif location == 'query':
+        raw_value = cast_param(request.params.get(param.name, None))
+    elif location == 'header':
+        raw_value = cast_param(request.headers.get(param.name, None))
+    elif location == 'formData':
+        if param_spec['type'] == 'file':
+            raw_value = request.params.get(param.name, None)
+        else:
+            raw_value = cast_param(request.params.get(param.name, None))
+    elif location == 'body':
+        # TODO: verify content-type header
+        raw_value = request.json()
+    else:
+        raise SwaggerMappingError(
+            "Don't know how to unmarshal_param with location {0}".
+            format(location))
+
+    if param_spec['type'] == 'array' and location != 'body':
+        raw_value = unmarshal_collection_format(param_spec, raw_value)
```

Comment:
It is a tricky list of conditionals to handle.  The orignial if/else is purposely only testing `location` for this reason.